### PR TITLE
Add 'any' and 'all' prebuilt function

### LINKED ISSUE #5128: [Adaptive expressions] Any() function request
There are several occasions where I need to check if there is any value in an array that matches a condition. This is currently possible using the where() method, but would be cleaner if there were an any() method which just returned the boolean value.

Here is an example of a condition I'm using for an if/else block that is checking for the presence of any parameters in my parameters object:
```
count(where($parameters, p, exists(p))) == 0
```

It would be more readable like this:
```
not(any(where($parameters, p, exists(p))))
```

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Danieladu**
Fixes #5128

## Description
Add `any` and `all` prebuilt function for Collections.

## Examples
`any(createArray(1, 'cool'), item, isInteger(item))` -> `true`
`any(createArray('first', 'cool'), item => isInteger(item))` -> `false`
`all(createArray(1, 'cool'), item, isInteger(item))` -> `false`
`all(createArray(1, 2), item => isInteger(item))` -> `true`

### CODE REVIEW

**Comment by mrivera-ms** [comment]
@Danieladu, is there an issue associated with this change? If so, please link it. Thank you.

File: libraries/AdaptiveExpressions/BuiltinFunctions/All.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            string error;
+
+            object instance;
+            (instance, error) = expression.Children[0].TryEvaluate(state, options);
+            if (error == null)
+            {
+                var list = FunctionUtils.ConvertToList(instance);
+                if (list == null)
+                {
+                    error = $"{expression.Children[0]} is not a collection or structure object to run any";
```
Comment (mrivera-ms) [inline libraries/AdaptiveExpressions/BuiltinFunctions/All.cs:33]: Should say "... to run All"

File: libraries/AdaptiveExpressions/BuiltinFunctions/Any.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            string error;
+
+            object instance;
+            (instance, error) = expression.Children[0].TryEvaluate(state, options);
+            if (error == null)
+            {
+                var list = FunctionUtils.ConvertToList(instance);
+                if (list == null)
+                {
+                    error = $"{expression.Children[0]} is not a collection or structure object to run any";
```
Comment (mrivera-ms) [inline libraries/AdaptiveExpressions/BuiltinFunctions/Any.cs:33]: Capitalize "any".

File: libraries/AdaptiveExpressions/BuiltinFunctions/Where.cs [hunk 3]
```diff
@@ -30,51 +30,26 @@ private static (object value, string error) Evaluator(Expression expression, IMe
...
-                    list = ilist;
-                }
-                else if (instance is JObject jobj)
-                {
-                    list = FunctionUtils.Object2KVPairList(jobj);
-                }
-                else if (FunctionUtils.ConvertToJToken(instance) is JObject jobject)
-                {
-                    list = FunctionUtils.Object2KVPairList(jobject);
+                    error = $"{expression.Children[0]} is not a collection or structure object to run where";
```
Comment (mrivera-ms) [inline libraries/AdaptiveExpressions/BuiltinFunctions/Where.cs:36]: Capitalize "Where"

File: libraries/AdaptiveExpressions/FunctionUtils.cs [hunk 4]
```diff
@@ -1050,18 +1073,18 @@ internal static List<object> Object2KVPairList(JObject jobj)
             return tempList;
         }
 
-        internal static void ValidateForeach(Expression expression)
+        internal static void ValidateLambdaExpression(Expression expression)
         {
             if (expression.Children.Length != 3)
             {
-                throw new ArgumentException($"foreach expects 3 parameters, found {expression.Children.Length}");
+                throw new ArgumentException($"lambda expression expects 3 parameters, found {expression.Children.Length}");
```
Comment (mrivera-ms) [inline libraries/AdaptiveExpressions/FunctionUtils.cs:1080]: Capitalize the first letter of the error message.

**Review by mrivera-ms — CHANGES_REQUESTED** [review]

**Review by luhan2017 — APPROVED** [review]

**Review by mrivera-ms — APPROVED** [review]
:shipit:

**Pull request merged**